pyaedt/filtersolutions_core/ideal_response.py

```diff
@@ -0,0 +1,458 @@
+    """
+
+    STEP_RESPONSE = 0
+    RAMP_RESPONSE = 1
+    IMPULSE_RESPONSE = 2
+    STEP_RESPONSE_DB = 3
+    RAMP_RESPONSE_DB = 4
+    IMPULSE_RESPONSE_DB = 5
+    TIME = 6
+
+
+class SParametersResponseColumn(Enum):
+    """Provides an enum of S parameters.
+
+    Attributes:
+    S21_DB: Represents the S21 parameter in dB.
+    S11_DB: Represents the S11 parameter in dB.
+    S21_ARITH: Represents the S21 parameter.
+    S11_ARITH: Represents the S11 parameter.
+    FREQUENCY: Represents the S parameters frequency parameter.
+    """
+
+    S21_DB = 0
+    S11_DB = 1
+    S21_ARITH = 3
+    S11_ARITH = 4
+    FREQUENCY = 5
+
+
+class PoleZerosResponseColumn(Enum):
+    """Provides an enum of pole zero x and y coordinates of transmission (TX) or reflection (RX) zeros.
+
+    Attributes:
+    TX_ZERO_DEN_X: Represents the x coordinate of filter transmission zero denominator.
+    TX_ZERO_DEN_Y: Represents the y coordinate of filter transmission zero denominator.
+    PROTO_TX_ZERO_DEN_X: Represents the x coordinate of prototype filter transmission zero denominator.
+    PROTO_TX_ZERO_DEN_Y: Represents the y coordinate of prototype filter transmission zero denominator.
+    TX_ZERO_NUM_X: Represents the x coordinate of filter transmission zero numerator.
+    TX_ZERO_NUM_Y: Represents the y coordinate of filter transmission zero numerator.
+    PROTO_TX_ZERO_NUM_X: Represents the x coordinate of prototype filter transmission zero numerator.
+    PROTO_TX_ZERO_NUM_Y: Represents the y coordinate of prototype filter transmission zero numerator.
+    RX_ZERO_DEN_X: Represents the x coordinate of filter reflection zero denominator.
+    RX_ZERO_DEN_Y: Represents the y coordinate of filter reflection zero denominator.
+    PROTO_RX_ZERO_DEN_X: Represents the x coordinate of prototype filter reflection zero denominator.
+    PROTO_RX_ZERO_DEN_Y: Represents the y coordinate of prototype filter reflection zero denominator.
+    RX_ZERO_NUM_X: Represents the x coordinate of filter reflection zero numerator.
+    RX_ZERO_NUM_Y: Represents the y coordinate of filter reflection zero numerator.
+    PROTO_RX_ZERO_NUM_X: Represents the x coordinate of prototype filter reflection zero numerator.
+    PROTO_RX_ZERO_NUM_Y: Represents the y coordinate of prototype filter reflection zero numerator.
+    """
+
+    TX_ZERO_DEN_X = 0
+    TX_ZERO_DEN_Y = 1
+    PROTO_TX_ZERO_DEN_X = 2
+    PROTO_TX_ZERO_DEN_Y = 3
+    TX_ZERO_NUM_X = 4
+    TX_ZERO_NUM_Y = 5
+    PROTO_TX_ZERO_NUM_X = 6
+    PROTO_TX_ZERO_NUM_Y = 7
+    RX_ZERO_DEN_X = 8
+    RX_ZERO_DEN_Y = 9
+    PROTO_RX_ZERO_DEN_X = 10
+    PROTO_RX_ZERO_DEN_Y = 11
+    RX_ZERO_NUM_X = 12
+    RX_ZERO_NUM_Y = 13
+    PROTO_RX_ZERO_NUM_X = 14
+    PROTO_RX_ZERO_NUM_Y = 15
+
+
+class IdealResponse:
+    """Returns the data for available ideal filter responses.
+    Includes ``frequency``, ``time``, ``S parameters``, ``transfer function``, or ``pole zero location`` responses.
```

Comment:
```suggestion

        Types of responses Include ``frequency``, ``time``, ``S parameters``, ``transfer function``, and ``pole zero location``.
```
Blank line needed before more detailed description.